src/ORM/DataObjectSchema.php

```diff
@@ -138,11 +138,11 @@ public function tableName($class)
      */
     public function baseDataClass($class)
     {
-        $class = ClassInfo::class_name($class);
```

Comment:
Should we not add the `if (!is_subclass_of($tableClass, DataObject::class))` check here?

Reply:
It does handle that at the end of the method, but I suppose we could move it to the start before the `while` loop